README.rst
Python Client for Binary Authorization API
==========================================

|stable| |pypi| |versions|

`Binary Authorization API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-binary-authorization.svg
   :target: https://pypi.org/project/google-cloud-binary-authorization/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-binary-authorization.svg
   :target: https://pypi.org/project/google-cloud-binary-authorization/
.. _Binary Authorization API: https://cloud.google.com/binary-authorization
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/binaryauthorization/latest
.. _Product Documentation:  https://cloud.google.com/binary-authorization

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Binary Authorization API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Binary Authorization API.:  https://cloud.google.com/binary-authorization
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-binary-authorization


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-binary-authorization

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Binary Authorization API
   to see other available methods on the client.
-  Read the `Binary Authorization API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Binary Authorization API Product documentation:  https://cloud.google.com/binary-authorization
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [0.3.1](https://www.github.com/googleapis/python-binary-authorization/compare/v0.3.0...v0.3.1) (2021-07-26)


### Features

* add Samples section to CONTRIBUTING.rst ([#56](https://www.github.com/googleapis/python-binary-authorization/issues/56)) ([c641b6c](https://www.github.com/googleapis/python-binary-authorization/commit/c641b6c148779e1952149e5ce5edf62fa0a1c642))


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#55](https://www.github.com/googleapis/python-binary-authorization/issues/55)) ([0ca0dc2](https://www.github.com/googleapis/python-binary-authorization/commit/0ca0dc2671bb8920f56bcbd057b9a13d7b23bf7f))
* enable self signed jwt for grpc ([#61](https://www.github.com/googleapis/python-binary-authorization/issues/61)) ([1a65f17](https://www.github.com/googleapis/python-binary-authorization/commit/1a65f171f677b7ca659ffe98051f432bed342209))


### Miscellaneous Chores

* release 0.3.1 ([#60](https://www.github.com/googleapis/python-binary-authorization/issues/60)) ([e2b54b5](https://www.github.com/googleapis/python-binary-authorization/commit/e2b54b5a97f23c6a01bce151b4fb5809f089f1d6))

## [0.3.0](https://www.github.com/googleapis/python-binary-authorization/compare/v0.2.2...v0.3.0) (2021-06-30)


### Features

* add always_use_jwt_access ([#48](https://www.github.com/googleapis/python-binary-authorization/issues/48)) ([63a3c9a](https://www.github.com/googleapis/python-binary-authorization/commit/63a3c9a8f8c9ab97436882adc7658260aa66df9d))


### Bug Fixes

* disable always_use_jwt_access ([#52](https://www.github.com/googleapis/python-binary-authorization/issues/52)) ([b840980](https://www.github.com/googleapis/python-binary-authorization/commit/b84098014328d14531caafe30585a5bd55c216f4))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-binary-authorization/issues/1127)) ([#43](https://www.github.com/googleapis/python-binary-authorization/issues/43)) ([726d589](https://www.github.com/googleapis/python-binary-authorization/commit/726d58920de4e97a70cbbe1fd88ac427224ba1ea)), closes [#1126](https://www.github.com/googleapis/python-binary-authorization/issues/1126)

### [0.2.2](https://www.github.com/googleapis/python-binary-authorization/compare/v0.2.1...v0.2.2) (2021-06-16)


### Bug Fixes

* exclude docs and tests from package ([#39](https://www.github.com/googleapis/python-binary-authorization/issues/39)) ([a90d7f4](https://www.github.com/googleapis/python-binary-authorization/commit/a90d7f46ca54c3bf805208bff157cfbc48a14234))

### [0.2.1](https://www.github.com/googleapis/python-binary-authorization/compare/v0.2.0...v0.2.1) (2021-05-25)


### Bug Fixes

* **deps:** add packaging requirement ([#34](https://www.github.com/googleapis/python-binary-authorization/issues/34)) ([59752a5](https://www.github.com/googleapis/python-binary-authorization/commit/59752a57cd6fb9a9e4d4caeb0b27793ce829d37c))

## [0.2.0](https://www.github.com/googleapis/python-binary-authorization/compare/v0.1.1...v0.2.0) (2021-05-20)


### Features

* Publish Binary Authorization ContinuousValidationEvent proto ([#31](https://www.github.com/googleapis/python-binary-authorization/issues/31)) ([d3d2abe](https://www.github.com/googleapis/python-binary-authorization/commit/d3d2abeb22bad714de0591916c1065fda7305a92))

### [0.1.1](https://www.github.com/googleapis/python-binary-authorization/compare/v0.1.0...v0.1.1) (2021-04-01)


### Bug Fixes

* use correct retry deadline ([#7](https://www.github.com/googleapis/python-binary-authorization/issues/7)) ([3f9bfc2](https://www.github.com/googleapis/python-binary-authorization/commit/3f9bfc2b1c5b6d520716b194daf175e1030135b0))


### Documentation

* update python contributing guide ([#9](https://www.github.com/googleapis/python-binary-authorization/issues/9)) ([b6e095f](https://www.github.com/googleapis/python-binary-authorization/commit/b6e095ff6a1f7422e9f1ce9132d32871f800aab7))

## 0.1.0 (2021-01-08)


### Features

* generate v1beta1 ([06c43f2](https://www.github.com/googleapis/python-binary-authorization/commit/06c43f24701da8f301be5bc04a6ec83a25edc41f))


owl-bot-staging/v1beta1/noxfile.py
# -*- coding: utf-8 -*-

# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

import os
import shutil

import nox  # type: ignore


@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/binaryauthorization_v1beta1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
